test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -151,8 +152,12 @@ func (r *DockerMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	if _, hasDeleteAnnotation := machine.Annotations[clusterv1.DeleteMachineAnnotation]; hasDeleteAnnotation {
+		dockerMachine.Annotations[clusterv1.DeleteMachineAnnotation] = machine.Annotations[clusterv1.DeleteMachineAnnotation]
+	}
+
 	// Create a helper for managing the docker container hosting the machine.
-	externalMachine, err := docker.NewMachine(ctx, cluster, machine.Name, nil)
+	externalMachine, err := docker.NewMachine(ctx, cluster, dockerMachine.Name, nil)
```

Comment:
The third parameter here is the name of the Docker container to create, and before we were using the name of the clusterv1.Machine. Since the Docker container is created before the DockerMachine, the DockerMachine needs to know what the name of the existing container is. Before, I had that in a field called `DockerMachine.Spec.InstanceName`, but I changed it so that the DockerMachine will have the same name as the Docker container.

Reply:
Let's keep using the clusterv1.Machine name here, this makes not relevant if you create first the container or the DockerMachine.

Reply:
I'm not sure I understand. Do you want me to have the owner Machines be created with the same name as the container? If so, that would require a change to the MP controller in CAPI core and outside of the CAPD scope.

Reply:
I've left it as is but left a more detailed comment in the new changes explaining what I'm trying to do. Feel free to start a new thread as this one is quite old.